app/services/invoices/payments/stripe_service.rb

```diff
@@ -51,6 +51,17 @@ def create
 
         result.payment = payment
         result
+      rescue Stripe::AuthenticationError, Stripe::CardError, Stripe::InvalidRequestError, Stripe::PermissionError => e
+        # NOTE: Do not mark the invoice as failed if the amount is too small for Stripe
+        #       For now we keep it as pending, the user can still update it manually
+        return if e.code == 'amount_too_small'
+
+        deliver_error_webhook(e)
+        update_invoice_payment_status(payment_status: :failed, deliver_webhook: false)
+        nil
+      rescue Stripe::RateLimitError, Stripe::APIConnectionError => e
+        deliver_error_webhook(e)
+        raise
```

Comment:
Don't we want to keep handling `Stripe::StripeError` and just "ignore" the `Stripe::RateLimitError`, `Stripe::APIConnectionError`?

```suggestion
      rescue Stripe::RateLimitError, Stripe::APIConnectionError => e
        raise # Let the auto-retry process do its own job
      rescue Stripe::StripeError => e
        deliver_error_webhook(e)
        raise
       end
```